components/ui/Alert.tsx

```diff
@@ -19,47 +19,143 @@
  */
 import * as classNames from 'classnames';
 import * as React from 'react';
+import styled from 'styled-components';
 import { translate } from '../../helpers/l10n';
 import Tooltip from '../controls/Tooltip';
 import AlertErrorIcon from '../icons/AlertErrorIcon';
 import AlertSuccessIcon from '../icons/AlertSuccessIcon';
 import AlertWarnIcon from '../icons/AlertWarnIcon';
 import InfoIcon from '../icons/InfoIcon';
-import ThemeContext from '../ThemeContext';
-import './Alert.css';
+import { SonarUiCommonTheme, SonarUiCommonThemeConsumer } from '../theme';
 
-type AlertDisplay = 'banner' | 'block' | 'inline';
+type AlertDisplay = 'banner' | 'inline' | 'block';
 type AlertVariant = 'error' | 'warning' | 'success' | 'info';
 
-export interface AlertProps {
+export interface Props {
   display?: AlertDisplay;
   variant: AlertVariant;
 }
 
-export function Alert(props: AlertProps & React.HTMLAttributes<HTMLDivElement>) {
+export function Alert(props: Props & React.HTMLAttributes<HTMLDivElement>) {
+  const isInline = () => props.display === 'inline';
+  const isBanner = () => props.display === 'banner';
+  const getAlertVariantInfo = (theme: SonarUiCommonTheme) => {
+    switch (props.variant) {
+      case 'error': {
+        return {
+          icon: <AlertErrorIcon fill={theme.colors.alertIconError} />,
+          color: theme.colors.alertTextError,
+          borderColor: theme.colors.alertBorderError,
+          backGroundColor: theme.colors.alertBackgroundError
+        };
+      }
+      case 'warning': {
+        return {
+          icon: <AlertWarnIcon fill={theme.colors.alertIconWarning} />,
+          color: theme.colors.alertTextWarning,
+          borderColor: theme.colors.alertBorderWarning,
+          backGroundColor: theme.colors.alertBackgroundWarning
+        };
+      }
+      case 'success': {
+        return {
+          icon: <AlertSuccessIcon fill={theme.colors.alertIconSuccess} />,
+          color: theme.colors.alertTextSuccess,
+          borderColor: theme.colors.alertBorderSuccess,
+          backGroundColor: theme.colors.alertBackgroundSuccess
+        };
+      }
+      case 'info': {
+        return {
+          icon: <InfoIcon fill={theme.colors.alertIconInfo} />,
+          color: theme.colors.alertTextInfo,
+          borderColor: theme.colors.alertBorderInfo,
+          backGroundColor: theme.colors.alertBackgroundInfo
+        };
+      }
+      default:
+        throw new Error(`Unsupported alert variant : ${props.variant}`);
+    }
+  };
+
+  const AlertIcon = styled.div.attrs(p => ({ variantInfo: getAlertVariantInfo(p.theme) }))`
+    ${(p: any) => `
+    flex: 0 0 auto;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: calc(${isBanner() ? 2 : 4} * ${p.theme.sizes.gridSize});
+    ${!isBanner() ? 'border-right: 1px solid;' : ''};
+    border-color: ${p.variantInfo.borderColor};
+    `}
+  `;
+
+  const AlertContent = styled.div`
+    ${(p: any) => `
```

Comment:
I find this hard to understand :sob:. What is the meaning of `p` ?

Reply:
p is the props of the styled-components, which contains the global styled-component theme. See 2 lines below.